SequenceType/FigurateNumbers.py

```diff
@@ -0,0 +1,176 @@
+
+        for i in range(1, len(self.seq.ls)):
+            if (root_term_1 + i) ** 2 != self.seq.ls[i]:
+                return False
+
+        return True
+
+    def term_number(self, index):
+        return (math.sqrt(self.seq.ls[-1]) + (index - self.seq.size)) ** 2
+
+    def seq_str(self, n):
+        return n ** 2
+
+
+class TriangularNumberSequence(SequenceType):
+
+    def __init__(self, seq):
+        super(TriangularNumberSequence, self).__init__(seq)
+        self.n_last_term = 0  # nth term in Triangular Number Sequence for the last term of the testing sequence
+
+    def __bool__(self):
+        """ Checks whether sequence contains Triangular Numbers"""
+
+        # get the nth Triangular Number
+        self.n_last_term = math.floor(max(np.roots([1, 1, -(2 * self.seq.ls[-1])])))
+
+        # last term is not a triangular number
+        if (self.n_last_term * (self.n_last_term + 1)) / 2 != self.seq.ls[-1]:
+            return False
+
+        for i in range(1, self.seq.size):
+            if ((self.n_last_term - i) * (self.n_last_term - i + 1)) / 2 != self.seq.ls[self.seq.size - i - 1]:
+                return False
+        return True
+
+    @staticmethod
+    def nth_triangular_number(n):
+        """ Get the nth term triangular number """
+        return n * (n + 1) / 2
+
+    def term_number(self, index):
+        """ Get the ith number in the sequence """
+        req_n_term = self.n_last_term + (index - self.seq.size)
+        return TriangularNumberSequence.nth_triangular_number(req_n_term)
+
+    def seq_str(self, n):
+        return n * (n + 1) / 2
+
+
+class PentagonalNumberSequence(SequenceType):
+
+    def __init__(self, seq):
+        super(PentagonalNumberSequence, self).__init__(seq)
+        self.n_last_term = 0  # nth term in Pentagonal Number Sequence for the last term of the testing sequence
+
+    def __bool__(self):
+        """ Checks whether sequence contains Pentagonal Numbers"""
+
+        # get the nth Pentagonal Number
+        self.n_last_term = math.floor(max(np.roots([3, -1, -(2 * self.seq.ls[-1])])))
+
+        # last term is not a Pentagonal number
+        if (self.n_last_term * (3 * self.n_last_term - 1)) / 2 != self.seq.ls[-1]:
+            return False
+
+        for i in range(1, self.seq.size):
+            if ((self.n_last_term - i) * (3 * (self.n_last_term - i) - 1)) / 2 != self.seq.ls[self.seq.size - i - 1]:
+                return False
+        return True
+
+    @staticmethod
+    def nth_pentagonal_number(n):
+        """ Get the nth term pentagonal number """
+        return n * (3 * n - 1) / 2
+
+    def term_number(self, index):
+        """ Get the ith number in the sequence """
+        req_n_term = self.n_last_term + (index - self.seq.size)
+        return PentagonalNumberSequence.nth_pentagonal_number(req_n_term)
+
+    def seq_str(self, n):
+        return n * (3 * n - 1) / 2
+
+
+class HexagonalNumberSequence(SequenceType):
```

Comment:
This is an interesting idea, but I think (I could be wrong) that the main sequence finder actually has generalized logic that would've already been able to find the closed form of this case, and in fact any case that comes from any polynomial closed form solution. 
So for example if you inputted `1, 6, 15, 28, 45, 66, 91, 120, 153, 190, 231, 276, 325, 378, 435, 496, 561, 630, 703, 780, 861, 946`, then even without this method it should still be able to specify that the closed form is `1.5*n^2 - 0.5*n`

There might be some merit in making it go a step further and recognize that the resulting polynomial has a name, i.e. Hexagonal numbers. 
It would mean a bit of an extension on the displaying of the result. What do you think?

Reply:
I really like that idea!  I'll take a stab :)